pkg/syndesis/action/install.go

```diff
@@ -24,29 +31,45 @@ func (a *Install) Execute(syndesis *v1alpha1.Syndesis) error {
 
 	logrus.Info("Installing Syndesis resource ", syndesis.Name)
 
-	sa := newSyndesisServiceAccount()
-	setNamespaceAndOwnerReference(sa, syndesis)
-	// We don't replace the service account if already present, to let Kubernetes generate its tokens
-	err := sdk.Create(sa)
-	if err != nil && !k8serrors.IsAlreadyExists(err) {
+	token, err := installServiceAccount(syndesis)
+	if err != nil {
 		return err
 	}
 
-	token, err := serviceaccount.GetServiceAccountToken(sa.Name, syndesis.Namespace)
+	list, err := syndesistemplate.GetInstallResourcesAsRuntimeObjects(syndesis, syndesistemplate.InstallParams{
+		OAuthClientSecret: token,
+	})
 	if err != nil {
 		return err
 	}
 
-	list, err := syndesistemplate.GetInstallResources(syndesis, syndesistemplate.InstallParams{
-		OAuthClientSecret: token,
-	})
+	syndesisRoute, err := installSyndesisRoute(syndesis, list)
+	if err != nil {
+		return err
+	}
 
-	for _, obj := range list {
-		res, err := util.LoadKubernetesResource(obj.Raw)
+	if syndesis.Spec.RouteHostName == "" {
+		syndesis.Spec.RouteHostName = syndesisRoute.Spec.Host
```

Comment:
wonder whether we should set the host name here or only on the deep copy in `target` below ? Not sure whether how the policy is to change incoming parameter.